asv_bench/benchmarks/alignment.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+
+import xarray as xr
+
+from . import parameterized, requires_dask
+
+ntime = 365 * 30
+nx = 50
+ny = 50
+
+rs = np.random.RandomState(0)
+
+
+class Align:
+    def setup(self, *args, **kwargs):
+        data = rs.randn(ntime, nx, ny)
+        self.ds = xr.Dataset(
+            {"temperature": (("time", "x", "y"), data)},
+            coords={
+                "time": xr.date_range("2000", periods=ntime),
+                "x": np.arange(nx),
+                "y": np.arange(ny),
+            },
+        )
+        self.year = self.ds.time.dt.year
+        self.idx = rs.randint(low=0, high=ntime, size=ntime // 2)
```

Comment:
```suggestion
        self.idx = np.unique(rng.integers(low=0, high=ntime, size=ntime // 2))
```

Using unique should hopefully fix this:
```
Error: [ 75.83%] ··· ...AlignCFTime.time_not_aligned_random_integers             failed
[ 75.83%] ··· ======= ========
                join          
              ------- --------
               outer   failed 
               inner   failed 
                left   failed 
               right   failed 
              ======= ========

[ 75.83%] ···· For parameters: 'outer'
Error:                Traceback (most recent call last):
                 File "/home/runner/micromamba-root/envs/xarray-tests/lib/python3.10/site-packages/asv/benchmark.py", line 1293, in main_run_server
                   main_run(run_args)
                 File "/home/runner/micromamba-root/envs/xarray-tests/lib/python3.10/site-packages/asv/benchmark.py", line 1167, in main_run
                   result = benchmark.do_run()
                 File "/home/runner/micromamba-root/envs/xarray-tests/lib/python3.10/site-packages/asv/benchmark.py", line 573, in do_run
                   return self.run(*self._current_params)
                 File "/home/runner/micromamba-root/envs/xarray-tests/lib/python3.10/site-packages/asv/benchmark.py", line 669, in run
                   samples, number = self.benchmark_timing(timer, min_repeat, max_repeat,
                 File "/home/runner/micromamba-root/envs/xarray-tests/lib/python3.10/site-packages/asv/benchmark.py", line 705, in benchmark_timing
                   timing = timer.timeit(number)
                 File "/home/runner/work/xarray/xarray/asv_bench/.asv/env/df282ba4a530a0853b7f9108ec3ff02d/lib/python3.10/timeit.py", line 178, in timeit
                   timing = self.inner(it, self.timer)
                 File "<timeit-src>", line 6, in inner
                 File "/home/runner/micromamba-root/envs/xarray-tests/lib/python3.10/site-packages/asv/benchmark.py", line 627, in <lambda>
                   func = lambda: self.func(*param)
                 File "/home/runner/work/xarray/xarray/asv_bench/benchmarks/alignment.py", line 39, in time_not_aligned_random_integers
                   xr.align(self.ds, self.year_subset, join=join)
                 File "/home/runner/work/xarray/xarray/asv_bench/.asv/env/df282ba4a530a0853b7f9108ec3ff02d/lib/python3.10/site-packages/xarray/core/alignment.py", line 785, in align
                   aligner.align()
                 File "/home/runner/work/xarray/xarray/asv_bench/.asv/env/df282ba4a530a0853b7f9108ec3ff02d/lib/python3.10/site-packages/xarray/core/alignment.py", line 578, in align
                   self.reindex_all()
                 File "/home/runner/work/xarray/xarray/asv_bench/.asv/env/df282ba4a530a0853b7f9108ec3ff02d/lib/python3.10/site-packages/xarray/core/alignment.py", line 555, in reindex_all
                   self.results = tuple(
                 File "/home/runner/work/xarray/xarray/asv_bench/.asv/env/df282ba4a530a0853b7f9108ec3ff02d/lib/python3.10/site-packages/xarray/core/alignment.py", line 556, in <genexpr>
                   self._reindex_one(obj, matching_indexes)
                 File "/home/runner/work/xarray/xarray/asv_bench/.asv/env/df282ba4a530a0853b7f9108ec3ff02d/lib/python3.10/site-packages/xarray/core/alignment.py", line 540, in _reindex_one
                   dim_pos_indexers = self._get_dim_pos_indexers(matching_indexes)
                 File "/home/runner/work/xarray/xarray/asv_bench/.asv/env/df282ba4a530a0853b7f9108ec3ff02d/lib/python3.10/site-packages/xarray/core/alignment.py", line 506, in _get_dim_pos_indexers
                   indexers = obj_idx.reindex_like(aligned_idx, **self.reindex_kwargs)
                 File "/home/runner/work/xarray/xarray/asv_bench/.asv/env/df282ba4a530a0853b7f9108ec3ff02d/lib/python3.10/site-packages/xarray/core/indexes.py", line 507, in reindex_like
                   raise ValueError(
               ValueError: cannot reindex or align along dimension 'time' because the (pandas) index has duplicate values
```